app/user.py

```diff
@@ -0,0 +1,119 @@
+from flask import Flask, Blueprint
+from flask_restful import Api, Resource, reqparse, fields, marshal,inputs
+
+from models import User
+from run_app import app, db
+
+# Declare Blueprint
+user_blueprint = Blueprint('user_endpoint', __name__)
+auth_api = Api(user_blueprint)
+
+user_fields = {
+    'user_id': fields.Integer,
+    'username': fields.String,
+    'email_address': fields.String,
+    'password_hash': fields.String
+}
+
+class LoginAPI(Resource):
+    """ Creates endpoints for LoginAPI """
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('email', type=str, required=True, \
+        help='No email address provided', location='json')
+        self.reqparse.add_argument('password', type=str, required=True, \
+        help='No password provided', location='json')
+
+    def post(self):
+        """ Logs a user in """
+        args = self.reqparse.parse_args()
+
+        # fetch user data
+        check_user = User.query.filter_by(email_address=args['email']).first()
+
+        try:
+            if check_user:
+                if check_user.verify_pass(args['password']):
+                    give_token = check_user.generate_auth_token()
+
+                    json_response = {
+                        'message': 'Login Successful',
+                        'auth_token': give_token.decode()
+                    }
+                    return json_response, 200
+
+                else:
+                    json_response = {
+                        'error': 'Invalid Password'
+                    }
+                    return json_response, 401
+            else:
+                json_response = {
+                    'error': 'User with email {} does not exist'.format(args['email'])
+                }
+                return json_response, 404
+        except:
+            json_response = {
+                'error': 'Login Failed'
+            }
+            return json_response, 500
+
```

Comment:
There is nested if else statements from L35 - L54. Perhaps another way is to use `if not` to avoid that. 
   ```
        if not check_user:
           return json_response, 404
        if not check_user.verify_pass(args['password']):
          return json_response, 401
   ```